end_to_end_tests/golden-record/my_test_api_client/models/model_with_union_property.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict, Union
+
+import attr
+
+from ..models.an_enum import AnEnum
+from ..models.an_int_enum import AnIntEnum
+from ..types import UNSET, Unset
+
+
+@attr.s(auto_attribs=True)
+class ModelWithUnionProperty:
+    """  """
+
+    a_property: Union[Unset, AnEnum, AnIntEnum] = UNSET
+
+    def to_dict(self) -> Dict[str, Any]:
+        a_property: Union[Unset, AnEnum, AnIntEnum]
+        if isinstance(self.a_property, Unset):
+            a_property = UNSET
+        elif isinstance(self.a_property, AnEnum):
+            a_property = UNSET
+            if not isinstance(self.a_property, Unset):
+                a_property = self.a_property
+
+        else:
+            a_property = UNSET
+            if not isinstance(self.a_property, Unset):
+                a_property = self.a_property
+
+        field_dict = {}
+        if a_property is not UNSET:
+            field_dict["a_property"] = a_property
+
+        return field_dict
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]) -> "ModelWithUnionProperty":
+        def _parse_a_property(data: Any) -> Union[Unset, AnEnum, AnIntEnum]:
+            data = None if isinstance(data, Unset) else data
```

Comment:
Why `None` this instead of checking for `Unset` further down? Less template code when `nullables` are involved maybe?

Reply:
Yeah, this was to be able to reuse `construct` from template for the inner property without further modifying.

Most of the `construct`s just check `if {{ source }} is not None` and don't account for `UNSET`s.

I'll look a bit more to see if there is a better way.

Reply:
If there's not an easy fix it's not a huge deal. If you want to just regen the custom template e2e test so the checks pass I'll merge this.